src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests.vb

```diff
@@ -909,5 +909,42 @@ Option Strict On 'BIND:"Option Strict On"
 
             VerifyNoOperationTreeForTest(Of StatementSyntax)(source)
         End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <WorkItem(23283, "https://github.com/dotnet/roslyn/issues/23283")>
+        <Fact()>
+        Public Sub TestColorColorScenario()
```

Comment:
>TestColorColorScenario [](start = 19, length = 22)

Simply because this test passes, doesn't really mean the BoundTypeOrValueExpression node never escapes initial binding. For example, I see a special case for this node in SemanticModel. This means that, at least at some point, the node was escaping initial binding. A simple way to find cases like that would be to throw in operation factory for this node and run all tests with test hook enabled. If that doesn't work, we'll need to think more